### PR TITLE
Add AuroraDriveFixerSmart and DeadEntryCleaner utility scripts

### PR DESCRIPTION
**Pull request by EduDicaseGameplay**
## Summary

This pull request introduces two utility scripts for Aurora, designed to simplify maintenance and improve the overall user experience.

### AuroraDriveFixerSmart
Automatically detects storage devices and repairs common issues related to scan paths after drive changes, cloning, or storage migration. The script helps Aurora locate games correctly again without requiring users to manually recreate their scan paths.

**Key features:**
- Automatic drive detection
- Smart scan path repair
- Supports HDD and USB devices
- Safe operation (does not modify game files)

### DeadEntryCleaner
Scans Aurora's `content.db` database for invalid or orphaned entries ("dead entries") and removes them safely, keeping the database clean and preventing duplicate or broken game listings.

**Key features:**
- Detects orphaned database entries
- Removes only invalid records
- Preserves valid games and metadata
- Helps maintain a clean and organized library

## Testing

Both scripts have been thoroughly tested on **Aurora 0.7b.2** using multiple storage configurations (HDD and USB). The scripts were also tested with different game layouts and databases to ensure safe and reliable operation.

## Notes

These scripts are intended to automate common maintenance tasks while minimizing the risk of user error. They are designed to be safe, lightweight, and easy to use for both new and experienced Aurora users.

### CODE REVIEW

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    local success = 0
+    local failed = 0
+    local removed = 0
+
+    for _, row in pairs(rows) do
+
+        local exists = Sql.ExecuteFetchRows(
+            "SELECT id FROM titleupdates WHERE filename='"..row["FileName"]..
+            "' AND titleid='"..row["TitleId"]..
+            "' AND version='"..row["Version"]..
```
Comment (Swizzy) [inline UtilityScripts/AuroraDriveFixerSmart/Main.lua]: BaseVersion + MediaID are also important, TitleId isn't the only deciding factor - different region versions have different MediaID/BaseVersion.

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    -- BACKUP WARNING
+    --------------------------------------------------
+    local backupWarning = Script.ShowMessageBox(
+        "Important Warning",
+        "Before continuing, it is highly recommended to create a backup of the Aurora database.\n\n" ..
+        "Default location:\n" ..
+        "Data\\Databases\\content.db\n" ..
+        "or\n" ..
+        "User\\Data\\Databases\\content.db\n\n" ..
+        "Do you want to continue anyway?",
```
Comment (Swizzy) [inline UtilityScripts/AuroraDriveFixerSmart/Main.lua:201]: Maybe enforce making a backup of the DB in the script directory? that way it's safer :)

File: UtilityScripts/DeadEntryCleaner/Main.lua [hunk 3]
```diff
@@ -0,0 +1,586 @@
+scriptTitle = "Dead Entry Cleaner (BETA)"
+scriptAuthor = "Eduardo Henrique/Channel Edu Dicas e Gameplay"
+scriptVersion = "1.0"
+scriptDescription = "Removes ghost entries from Aurora (ContentItems). Does not alter games, DLCs, or TUs. May require a re-scan."
```
Comment (Swizzy) [inline UtilityScripts/DeadEntryCleaner/Main.lua:4]: You can trigger a Rescan/Restart if memory serves me right, check some of the other scripts we've made - there should be something in there for that.

File: UtilityScripts/DeadEntryCleaner/Main.lua [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+
+function safeLower(text)
+    if not text then return "" end
+    return string.lower(tostring(text))
+end
+
+function isValidTitleId(id)
+    if not id then return false end
+    if string.len(id) ~= 8 then return false end
+    if id == "00000000" then return false end
```
Comment (Swizzy) [inline UtilityScripts/DeadEntryCleaner/Main.lua]: Homebrew might have all 0's as the TitleID, don't make it a true rule.

File: UtilityScripts/DeadEntryCleaner/Main.lua [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+        local serial = d["Serial"] or "UNKNOWN"
+
+        local label = ""
+
+        if string.find(safeLower(mount), "hdd") then
+            label = "HDD (Internal Storage)"
+        elseif string.find(safeLower(mount), "usb") then
+            label = "USB (External Device)"
+        else
+            label = "Storage Device"
```
Comment (Swizzy) [inline UtilityScripts/DeadEntryCleaner/Main.lua]: As a fallback you might want to use "Storage device (" .. mount .. ")" instead when unknown?

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua (hunk 1 above)
Comment (EduDicaseGameplay) [reply UtilityScripts/AuroraDriveFixerSmart/Main.lua]: That's exactly why I implemented it this way: to ensure it doesn't ignore different versions. If it relied only on the **Title ID** and **Media ID**, it would end up treating different versions as if they were the same. That's why I decided to make the **version** a more decisive factor in the identification process.

For example, a **Title Update** could be **TU10**, but it might be a modified build. In that case, the **Title ID**, **Media ID**, and even the **TU number** would remain the same, while the **version** would change, allowing the files to be distinguished correctly.

In my tests with several games, it behaved as expected every time and worked correctly in all cases. Of course, that doesn't mean there couldn't be a specific edge case where it fails, but so far all of my tests have been successful.

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua (hunk 2 above)
Comment (EduDicaseGameplay) [reply UtilityScripts/AuroraDriveFixerSmart/Main.lua:201]: Well, I thought it was better not to enforce it since it's not mandatory. However, reminding users to make a backup before making any changes is never a bad idea.

Unfortunately, I tried several different approaches, but I couldn't get the database to create a backup while it was in use. Because of that, I felt the best alternative was to simply display a clear warning and recommend that the user create a backup before proceeding.

File: UtilityScripts/DeadEntryCleaner/Main.lua (hunk 4 above)
Comment (EduDicaseGameplay) [reply UtilityScripts/DeadEntryCleaner/Main.lua]: Well, I really can't treat that as an absolute. I'll look into implementing a better solution as soon as I have some free time.

File: UtilityScripts/DeadEntryCleaner/Main.lua (hunk 5 above)
Comment (EduDicaseGameplay) [reply UtilityScripts/DeadEntryCleaner/Main.lua]: I can certainly look into that change. So far, though, it hasn't caused any issues in my testing. That said, all suggestions are welcome, and I'll be happy to evaluate whether it can be implemented.

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua (hunk 1 above)
Comment (Swizzy) [reply UtilityScripts/AuroraDriveFixerSmart/Main.lua]: BaseVersion is not the same thing as Version, and i don't see you including MediaID either in the uniqueness check.

You are correct that it is a edge case, you need 2 or more different versions of the same game for this to actually happen, that said it should be handled correctly.

Modified TUs won't have a different Version either, they will simply have a different hash.

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua (hunk 2 above)
Comment (Swizzy) [reply UtilityScripts/AuroraDriveFixerSmart/Main.lua:201]: Makes sense :)

File: UtilityScripts/AuroraDriveFixerSmart/Main.lua (hunk 1 above)
Comment (EduDicaseGameplay) [reply UtilityScripts/AuroraDriveFixerSmart/Main.lua]: I’m going to look into this to get the necessary information and try to improve it; I relied heavily on the tables to reach this result, and it worked in my tests, but—as you mentioned—it might fail in certain specific cases. When I get home, I’ll try to fix it: I’ll check the `content.db` database and the `titleupdates` table to see if those specific columns exist, and if they do, I’ll add them to the analysis.
